src/JBrowse/View/Track/BlockBased.js

```diff
@@ -861,27 +861,45 @@ return declare( [Component,DetailsMixin,FeatureFiltererMixin,Destroyable],
         var fmt = lang.hitch(this, 'renderDetailField', details );
         fmt( 'Name', this.key || this.name );
         var metadata = lang.clone( this.getMetadata() );
-        lang.mixin( metadata, additional );
         delete metadata.key;
         delete metadata.label;
         if( typeof metadata.conf == 'object' )
             delete metadata.conf;
-
+        if (this.browser && this.browser.config && this.browser.config.trackSelector && this.browser.config.trackSelector.renameFacets){
+           var metadataCopy = {};
+           for (var k in metadata){
+              key = this.browser.config.trackSelector.renameFacets[k] || k;
```

Comment:
`key` isn't defined, which is why the build is not completing. could you change that line to `let key = ...`?